update

NemesisWar opened the pull request:
update project

NemesisWar committed:
update

update project

NemesisWar committed:
repair some bags

HolyMonkey commented on Assets/Scripts/Camera/CameraMove.cs:
```diff
@@ -0,0 +1,25 @@
+    private void Start()
+    {
+        _camera = GetComponent <Camera>();
+        _cameraPosition = _camera.transform.position;
```
?????

HolyMonkey commented on Assets/Scripts/Camera/CameraMove.cs:
```diff
@@ -0,0 +1,25 @@
+    [SerializeField] private Transform _player;
+    private Camera _camera;
+    private Vector3 _cameraPosition;
+    private float _distanseBetweenPlayerAndCameraX;
```
_offset

HolyMonkey commented on Assets/Scripts/Camera/CameraMove.cs:
```diff
@@ -0,0 +1,25 @@
+{
+    [SerializeField] private Transform _player;
+    private Camera _camera;
+    private Vector3 _cameraPosition;
```
зачем это всё в поле хранится?

HolyMonkey commented on Assets/Scripts/Camera/CameraMove.cs:
```diff
@@ -0,0 +1,25 @@
+public class CameraMove : MonoBehaviour
+{
+    [SerializeField] private Transform _player;
+    private Camera _camera;
```
зачем тебе камера?

HolyMonkey commented on Assets/Scripts/Map/AutoDeleteMap.cs:
```diff
@@ -0,0 +1,24 @@
+
+public class AutoDeleteMap : MonoBehaviour
+{
+    public UnityAction<Vector2Int,GameObject> DeleteObject;
```
https://trello.com/c/CsOZfMhi/1-%D0%BD%D0%B0%D0%B7%D0%B2%D0%B0%D0%BD%D0%B8%D1%8F-%D1%81%D0%BE%D0%B1%D1%8B%D1%82%D0%B8%D0%B9-%D0%BE%D0%B1%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D1%87%D0%B8%D0%BA%D0%BE%D0%B2

И ты куда-то потерял слово event

HolyMonkey commented on Assets/Scripts/Map/AutoDeleteMap.cs:
```diff
@@ -0,0 +1,24 @@
+using UnityEngine;
+using UnityEngine.Events;
+
+public class AutoDeleteMap : MonoBehaviour
```
Нужно образовать существительное

HolyMonkey commented on Assets/Scripts/Map/AutoDeleteMap.cs:
```diff
@@ -0,0 +1,24 @@
+    {
+        if (Vector2.Distance(transform.position,_player.position) > 21)
+        {
+            DeleteObject?.Invoke(new Vector2Int((int)transform.position.x, (int)transform.position.y),gameObject);
```
Событие сообщает о том, что что-то случилось а не о том, что что-то надо сделать. 
Для этого обратись к тому, что делает напрямую

HolyMonkey commented on Assets/Scripts/Map/AutoDeleteMap.cs:
```diff
@@ -0,0 +1,24 @@
+
+    private void FixedUpdate()
+    {
+        if (Vector2.Distance(transform.position,_player.position) > 21)
```
Почему 21? Это магическое число

HolyMonkey commented on Assets/Scripts/Map/Coin.cs:
```diff
@@ -0,0 +1,15 @@
+    {
+        if (collision.TryGetComponent(out Player player))
+        {
+            player.ChangeCoins();
```
Change слишком общий глагол, тут скорей Add

HolyMonkey commented on Assets/Scripts/Map/GenerateMap.cs:
```diff
@@ -0,0 +1,89 @@
+using UnityEngine;
+using System.Linq;
+
+public class GenerateMap : MonoBehaviour
```
Не используй мой код, ты внёс в него свои правки и полностью сломал весь концепт. Напиши свой генератор

NemesisWar replied:
Полностью уйти от твоего концепта не удалось. Изменил скрипт. Сделал его более понятней для себя.

HolyMonkey replied:
>Полностью уйти от твоего концепта не удалось

А придётся

NemesisWar replied:
Готово)

HolyMonkey commented on Assets/Scripts/UI/Paralax.cs:
```diff
@@ -0,0 +1,43 @@
+    //{
+    //    _imagePositionX += _speed * Time.deltaTime;
+    //    _image.uvRect = new Rect(_imagePositionX, 0, _image.uvRect.width, _image.uvRect.height);
+    //}
```
Зачем этот мусор в репозитории?

NemesisWar committed:
made changes according to the recommendations

HolyMonkey commented on Assets/Scripts/Map/Cleaner.cs:
```diff
@@ -0,0 +1,26 @@
+
+public class Cleaner : MonoBehaviour
+{
+    public event UnityAction<Vector2Int, GameObject> ExcessDistance;
```
https://stackoverflow.com/questions/150479/order-of-items-in-classes-fields-properties-constructors-methods
https://trello.com/b/aerR0R28/unity

Почитай как называются события

NemesisWar replied:
Скорректировал на ObjectNotVisible

HolyMonkey commented on Assets/Scripts/Map/Cleaner.cs:
```diff
@@ -0,0 +1,26 @@
+    private int _deleteRadius = 1;
+    private Transform _player;
+
+    public void Init(Transform player, int vievRadius)
```
viewRadius

NemesisWar replied:
Переименовал

HolyMonkey commented on Assets/Scripts/Map/Cleaner.cs:
```diff
@@ -0,0 +1,26 @@
+
+    private void FixedUpdate()
+    {
+        if (Vector2.Distance(transform.position, _player.position) > _deleteRadius)
```
Для каждого объекта дистанцию мерить очень дорого

NemesisWar replied:
Прибегнул к вычитанию позиции только по одной оси.

NemesisWar committed:
Changed

NemesisWar committed:
changed

According to the StyleCop Rules(1201,1202,1203).
Added space between fields and events

HolyMonkey commented on Assets/Scripts/Map/Cleaner.cs:
```diff
@@ -0,0 +1,27 @@
+    private int _viewRadius = 1;
+    private Transform _player;
+
+    public event UnityAction<Vector2Int, GameObject> ObjectNotVisible;
```
Имя событие должно отражать что объект СТАЛ невидимым а не невидим всегда. По сути сейчас имя говорит, что будет сообщать о всех невидимых объектах примерно каждый кадр

NemesisWar replied:
переименовал на BecameInvisible

HolyMonkey commented on Assets/Scripts/Map/GenerateMap.cs:
```diff
@@ -0,0 +1,77 @@
+
+    private void TrySpawn(Vector2Int spawnPoint)
+    {
+        foreach (var grid in _gridObjects)
```
это object а не grid

NemesisWar replied:
Именование поля изменено.

HolyMonkey commented on Assets/Scripts/Map/GenerateMap.cs:
```diff
@@ -0,0 +1,77 @@
+                createdObject.GetComponent<Cleaner>().Init(_playerMover.transform, _viewDistance);
+                createdObject.GetComponent<Cleaner>().ObjectNotVisible += OnObjectNotVisible;
+            }
+        }
```
Зачем тут цикл?

NemesisWar replied:
> Зачем тут цикл?

Убрал цикл. Хотел реализовать тот вариант, когда на данных координатах уже есть объект что бы он искал себе другое место.

HolyMonkey commented on Assets/Scripts/Map/GenerateMap.cs:
```diff
@@ -0,0 +1,77 @@
+                _spawnedPositions.Add(spawnPoint);
+                var createdObject = Instantiate(spawnObject, (Vector2)spawnPoint, Quaternion.identity, transform);
+                createdObject.GetComponent<Cleaner>().Init(_playerMover.transform, _viewDistance);
+                createdObject.GetComponent<Cleaner>().ObjectNotVisible += OnObjectNotVisible;
```
Дубляж кода. И зачем каждый объект связывать с игроком? Собери их все в отдельный список в одном экземпляре Cleaner

NemesisWar replied:
Сделал.

HolyMonkey commented on Assets/Scripts/Map/GenerateMap.cs:
```diff
@@ -0,0 +1,77 @@
+    private void OnObjectNotVisible(Vector2Int deletedObject, GameObject unsubscribeObject)
+    {
+        _spawnedPositions.Remove(deletedObject);
+        unsubscribeObject.GetComponent<Cleaner>().ObjectNotVisible -= OnObjectNotVisible;
```
Ну вот даже тут эта связка с тем компонентов выглядит ненужной

NemesisWar replied:
> Ну вот даже тут эта связка с тем компонентов выглядит ненужной

После того как создал отдельный экземпляр Cleaner - данная связка боле не потребовалась.

HolyMonkey commented on Assets/Scripts/Player/Player.cs:
```diff
@@ -0,0 +1,17 @@
+{
+    private int _coins;
+
+    public event UnityAction<int> ChangeCoinCount;
```
https://trello.com/c/CsOZfMhi/1-%D0%BD%D0%B0%D0%B7%D0%B2%D0%B0%D0%BD%D0%B8%D1%8F-%D1%81%D0%BE%D0%B1%D1%8B%D1%82%D0%B8%D0%B9-%D0%BE%D0%B1%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D1%87%D0%B8%D0%BA%D0%BE%D0%B2

NemesisWar replied:
> https://trello.com/c/CsOZfMhi/1-%D0%BD%D0%B0%D0%B7%D0%B2%D0%B0%D0%BD%D0%B8%D1%8F-%D1%81%D0%BE%D0%B1%D1%8B%D1%82%D0%B8%D0%B9-%D0%BE%D0%B1%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D1%87%D0%B8%D0%BA%D0%BE%D0%B2

Скорректировал название события на CoinCollecting

HolyMonkey commented on Assets/Scripts/Player/PlayerMover.cs:
```diff
@@ -0,0 +1,77 @@
+
+    private void Start()
+    {
+        _playerInput.PlayerAction.Jump.performed += ctx => OnJump();
```
А отписываться кто будет?

NemesisWar replied:
> А отписываться кто будет?

Поправил.

HolyMonkey commented on Assets/Scripts/Player/PlayerMover.cs:
```diff
@@ -0,0 +1,77 @@
+        _playerInput.Enable();
+    }
+
+    private void OnJump()
```
TryJump

NemesisWar replied:
> TryJump

Переименовал

HolyMonkey commented on Assets/Scripts/Player/PlayerMover.cs:
```diff
@@ -0,0 +1,77 @@
+        if (_blockMove == false)
+        {
+            PlayerMoved?.Invoke(_blockMove);
+            Vector2 move = new Vector2(transform.position.x + (Vector2.right.x * _speed * Time.deltaTime), transform.position.y);
```
var move

HolyMonkey replied:
var nextPosition даже наверное

NemesisWar replied:
> var nextPosition даже наверное

скорректировал

HolyMonkey commented on Assets/Scripts/Player/PlayerMover.cs:
```diff
@@ -0,0 +1,77 @@
+    private void FixedUpdate()
+    {
+        _onGround = CheckGround(Vector2.down);
+        _blockMove = CheckGround(Vector2.right);
```
зачем это в поле хранится?

NemesisWar replied:
> зачем это в поле хранится?

Удалил

HolyMonkey commented on Assets/Scripts/Player/PlayerMover.cs:
```diff
@@ -0,0 +1,77 @@
+
+    private void FixedUpdate()
+    {
+        _onGround = CheckGround(Vector2.down);
```
зачем это в поле хранится?

NemesisWar replied:
> зачем это в поле хранится?

Удалил

HolyMonkey commented on Assets/Scripts/Player/PlayerMover.cs:
```diff
@@ -0,0 +1,77 @@
+            _rigidbody2D.position = move;
+            if ((int)_rigidbody2D.position.x > _previousPositionX)
+            {
+                _previousPositionX = (int)_rigidbody2D.position.x;
```
if((move - _rigidbody2D.position).sqrMagnitude != 0)

NemesisWar replied:
> if((move - _rigidbody2D.position).sqrMagnitude != 0)

Данный вариант не подошел, так как создание объектов карты происходит только после смещение по оси Х на одну координату. А использование данного решения будем вызывать создание объектов если игрок будет прыгать.
Попробовал использовать if((move.x - (int)_rigidbody2D.position.x)!=0) в этом случае он бывает пропускает создание новых объектов карты. Выявить проблему не удалось. Поэтому пришлось оставить переменную _previousPositionX. С ней код работает корректно.

HolyMonkey commented on Assets/Scripts/UI/MainMenu.cs:
```diff
@@ -0,0 +1,24 @@
+public class MainMenu : MonoBehaviour
+{
+    [SerializeField] private TMP_Text _developers;
+    public void LoadScene()
```
нужна строка разделитель

NemesisWar replied:
> нужна строка разделитель

поправил

HolyMonkey commented on Assets/Scripts/UI/MainMenu.cs:
```diff
@@ -0,0 +1,24 @@
+    [SerializeField] private TMP_Text _developers;
+    public void LoadScene()
+    {
+        SceneManager.LoadScene("Game");
```
https://trello.com/c/0fV5bxqA/13-%D0%B7%D0%B0%D0%B3%D1%80%D1%83%D0%B7%D0%BA%D0%B0-%D1%81%D1%86%D0%B5%D0%BD

NemesisWar replied:
> https://trello.com/c/0fV5bxqA/13-%D0%B7%D0%B0%D0%B3%D1%80%D1%83%D0%B7%D0%BA%D0%B0-%D1%81%D1%86%D0%B5%D0%BD

Сделал через плагин

NemesisWar committed:
update

HolyMonkey commented on Assets/Scripts/Camera/CameraMove.cs:
```diff
@@ -0,0 +1,20 @@
+using UnityEngine;
+
+[RequireComponent(typeof(Camera))]
+public class CameraMove : MonoBehaviour
```
PlayerTracker
FollowedCamera 
PlaterFollower

HolyMonkey commented on Assets/Scripts/Map/Cleaner.cs:
```diff
@@ -0,0 +1,57 @@
+        List<GridObject> removableObjects = new List<GridObject>();
+        foreach (var removableObject in _mapObjects)
+        {
+            if ((playerPositionX- removableObject.transform.position.x) > _viewRadius)
```
playerPositionX - removableObject.transform.position.x

HolyMonkey commented on Assets/Scripts/Map/GenerateMap.cs:
```diff
@@ -0,0 +1,79 @@
+    private void Spawn(GridObject spawnObject, Vector2Int spawnPoint)
+    {
+        spawnPoint.y = Random.Range(0, spawnObject.MaxLayer);
+        if (_spawnedPositions.Contains(spawnPoint))
+        {
+            return;
+        }
```
if(..)
  return 
  
  //code

HolyMonkey commented on Assets/Scripts/Map/GenerateMap.cs:
```diff
@@ -0,0 +1,79 @@
+    {
+        foreach (var mapObject in _gridObjects)
+        {
+            if (CanSpawn(mapObject) == true)
+            {
+                Spawn(mapObject, spawnPoint);
+            }
```
Tester-Doer нужен для того, чтобы проверить возможность операции. 
У тебя же Spawn можем не сработать даже если CanSpawn вернёт true

HolyMonkey commented on Assets/Scripts/Map/GenerateMap.cs:
```diff
@@ -0,0 +1,79 @@
+        TrySpawn(new Vector2Int(playerPositionX + _viewDistance, 0));
+    }
+
+    private void TrySpawn(Vector2Int spawnPoint)
```
TrySpawn от Spawn отличается не только воможность ошибки, что означает что это разные методы и должны называться по разному

HolyMonkey commented on Assets/Scripts/Player/Player.cs:
```diff
@@ -0,0 +1,17 @@
+
+    public event UnityAction<int> CoinCollected;
+
+    public void AddCoins()
```
AddCoin

HolyMonkey commented on Assets/Scripts/Player/PlayerMover.cs:
```diff
@@ -0,0 +1,74 @@
+[RequireComponent(typeof(Rigidbody2D))]
+public class PlayerMover : MonoBehaviour
+{
+    [SerializeField] private float _rayDistance;
```
Ну и как понять что это за ray такой?

HolyMonkey commented on Assets/Scripts/Player/PlayerMover.cs:
```diff
@@ -0,0 +1,74 @@
+        }
+    }
+
+    private bool CheckGround(Vector2 rayVector)
```
Grounded
IsGrounded

HolyMonkey commented on Assets/Scripts/Player/PlayerMover.cs:
```diff
@@ -0,0 +1,74 @@
+
+    private void FixedUpdate()
+    {
+        if (CheckGround(Vector2.right) == false)
```
Это проверка на то, что мы в полёте?
